src/sas/qtgui/Utilities/TabbedModelEditor.py

```diff
@@ -299,6 +299,36 @@ def updateFromPlugin(self):
         self.parent.communicate.statusBarUpdateSignal.emit(msg)
         logging.info(msg)
 
+    def checkModel(self, model_str):
+        """
+        Run the ast check
+        and return True if the model is good.
+        False otherwise.
+        """
+        successfulCheck = True
+        try:
+            ast.parse(model_str)
+
+        except SyntaxError as ex:
+            msg = "Error building model: " + str(ex)
+            logging.error(msg)
+            # print three last lines of the stack trace
+            # this will point out the exact line failing
+            last_lines = traceback.format_exc().split('\n')[-4:]
```

Comment:
Is there more in the traceback related to the type of error? The line number and line of code can be helpful, but not in all cases, e.g.
```
y = ((m*x + b)

return y
```
Gives the error:
```
ERROR: Error building model: invalid syntax (<unknown>, line 36)
ERROR:     return y
```

Reply:
I looked at various exceptions and it does not seem anything over the last 3 lines is all that useful.
E.g. the above gives

```
4:38:26 - ERROR: Error building model: invalid syntax (<unknown>, line 35)
14:38:26 - ERROR: Traceback (most recent call last):
  File "C:\Users\piotr\projects\sasview\src\sas\qtgui\Utilities\TabbedModelEditor.py", line 310, in checkModel
    ast.parse(model_str)
  File "C:\Anaconda3\envs\qt5_win\lib\ast.py", line 35, in parse
    return compile(source, filename, mode, PyCF_ONLY_AST)
  File "<unknown>", line 35
    return y
```

The extra information is about the AST module, so I think it would be confusing to the user.